Adding nine patch support to Android splashscreen generation (featur request #7751)

sluedecke opened the pull request:
Adds parameter --nine-patch which lets splashscreens for Android be generated
as 9-patch files if splashscreen-icon-ratio > 0.  Stretchable area must be defined in percentage
of width and height (e.g. --nine-patch 5,10 will result in 5% of image width left and 5%
of image width right, as well as 10% of image height both top and bottom to be stretchable)

Please see https://github.com/quasarframework/quasar/issues/7751 for a rationale.

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Documentation
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch and _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] It's been tested on a Cordova (iOS, Android) app
- [ ] It's been tested on a Electron app
- [ ] Any necessary documentation has been added or updated [in the docs](https://github.com/quasarframework/quasar/tree/dev/docs) (for faster update click on "Suggest an edit on GitHub" at bottom of page) or explained in the PR's description.

If adding a **new feature**, the PR's description includes:
- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

sluedecke committed:
Initial implementation for https://github.com/quasarframework/quasar/issues/7751

Adds parameter --nine-patch which lets splashscreens for Android be generated
as 9-patch files if splashscreen > 0.  The icon will be preserved from scaling.

sluedecke committed:
Improved implementation allowing to specify percentage of image area which is stretchable

sluedecke committed:
Documentation updates for #7751

sluedecke committed:
Set version to 2.3.4-cc to differentiate from upstream

rstoenescu commented:
Hi,

Thank you for working on this. You'll have to wait a bit for it to get merged as our current focus is on porting Quasar to Vue 3.

sluedecke committed:
Fixes:
- properly set edge pixels to be transparent as required from aapt
- properly dispose empty splash.png files left over from ensureFileSync

sluedecke commented:
Any time.  Please use latest commits on this branch if you consider integrating this - there are important fixes.

sluedecke committed:
FIX: Reviewed 9-patch specification and properly set content area (right and bottom edge of image)

sluedecke committed:
Should fix certain logos (mostly with white / grey .... is this really the cause or is it transparency?)

sluedecke committed:
Added --platform ios|android to filter cordova/capacitor icons
This enables the usage of different base icons per platform and
thus better support for different sizing and layout requirements
(see https://developer.android.com/guide/practices/ui_guidelines/icon_design_adaptive)
